Test/dafny0/AndOrAmbiguity.dfy

```diff
@@ -6,56 +6,56 @@
 // The tests below check that these ambiguities result in understandable error message.
 
 method M0() returns (A: bool, B: bool)
-  ensures A && B ||  // error: and-or ambiguity
+  ensures A && B || A // error: and-or ambiguity
 
 method M1() returns (A: bool, B: bool)
-  ensures A || B &&  // error: and-or ambiguity
+  ensures A || B && A // error: and-or ambiguity
 
 method M2() returns (A: bool, B: bool)
-  ensures && B ||  // error: and-or ambiguity
+  ensures && B || A // error: and-or ambiguity
 
 method M3() returns (A: bool, B: bool)
-  ensures || B &&  // error: and-or ambiguity
+  ensures || B && A // error: and-or ambiguity
 
 method M4() returns (A: bool, B: bool)
-  ensures && A && B ||  // error: and-or ambiguity
+  ensures && A && B || A // error: and-or ambiguity
 
 method M5() returns (A: bool, B: bool)
-  ensures || A || B &&  // error: and-or ambiguity
+  ensures || A || B && A // error: and-or ambiguity
 
 // ---
 
 method N0() returns (A: bool, B: bool, C: bool)
-  ensures A ==> B <==  // error: plication ambiguity
```

Comment:
Typo? _imp_ lication

Reply:
I see. "plication" as a general name for "_im_ plication `==>` or _ex_ plication `<==`". 😁

Reply:
Changed it to implication/explication ambiguity. No point in being cryptic and 'explication' is already a neologism (or at least an off-label use of the word) so 'plication' is a neoneologism

Reply:
Looks good now, thanks